forc-pkg/src/pkg.rs

```diff
@@ -732,6 +733,35 @@ pub(crate) fn fetch_deps(
     Ok((graph, path_map))
 }
 
+fn apply_patch(
+    name: &String,
+    source: &Source,
+    manifest: &Manifest,
+    parent_path: &Path,
+) -> Result<Source> {
+    let mut patches = manifest.patches();
+    match source {
+        //Check if the patch is for a git dependency.
+        Source::Git(git) => {
+            //Check if we got a patch for the git dependency.
+            if let Some(source_patches) = patches.find(|patch| *patch.0 == git.repo.to_string()) {
```

Comment:
Rather than iterating over all of the entries, `manifest.patch` is already a `BTreeMap` with log(n) lookup. We should use something like `manifest.patch.as_ref().and_then(|patches| patches.get(&git.repo.as_str()))` here.